File: Programmers/Dynamic_Programming/2xn_타일링.cpp
#include <string>
#include <vector>

using namespace std;

const int MOD = 1000000007;
int dp[60001];

int solution(int n) {
    if(n <= 2) {
        return dp[n] = n;
    }
    if(dp[n] > 0) {
        return dp[n];
    }
    return dp[n] = (solution(n-2) + solution(n-1)) % MOD;
}

File: Programmers/Dynamic_Programming/2xn_타일링.js
function solution(n) {
    const MOD = 1000000007;
    const dp = Array(60001).fill(0);
    
    for(let i=1; i <= n; i++) {
        if(i <= 2) {
            dp[i] = i;
        } else {
            dp[i] = (dp[i-1] + dp[i-2]) % MOD;   
        }
    }
    return dp[n];
}
